app/config/Config.scala

```diff
@@ -28,12 +29,18 @@ object Config extends AwsInstanceTags {
   }
   val config = ConfigurationMagic(appName, configMagicMode).load
 
-  def getOptionalProperty[T](path: String, getVal: String => T) = {
+  def getOptionalProperty[T](path: String, getVal: String => T): Option[T] = {
     if (config.hasPath(path)) Some(getVal(path))
     else None
   }
 
-  val elkKinesisStream = config.getString("elk.kinesis.stream")
+  def getPropertyIfEnabled(enabled: Boolean, path: String): String =
+    if (enabled) getOptionalProperty(path, config.getString).getOrElse(sys.error(s"Property $path is required"))
+    else s"feature requiring $path is disabled"
```

Comment:
I don't think so - it will only be used if the feature has been disabled via a config switch which isn't an error exactly